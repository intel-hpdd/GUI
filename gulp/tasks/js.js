//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
