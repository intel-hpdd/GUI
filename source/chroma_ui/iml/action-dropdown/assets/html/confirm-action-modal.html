<div class="modal-header">
  <h4 class="modal-title">Confirm {{ confirmAction.title }}</h4>
</div>
<div class="modal-body">
  <div class="single-confirmation" ng-if="confirmAction.confirmPrompts.length === 1">
    <p>{{ confirmAction.confirmPrompts[0] }}</p>
  </div>
  <div class="multi-confirmation" ng-if="confirmAction.confirmPrompts.length > 1">
    <h4>This action will:</h4>
    <ul class="well">
      <li ng-repeat="message in confirmAction.confirmPrompts">{{ message }}</li>
    </ul>
  </div>
</div>
<div class="modal-footer">
  <div class="btn-group" uib-dropdown>
    <button type="button" ng-click="$close(false)" class="btn btn-success">
      Confirm <i class="fa fa-check-circle-o"></i>
    </button>
    <button type="button" class="btn btn-success" uib-dropdown-toggle>
      <span class="caret"></span>
      <span class="sr-only">Split button</span>
    </button>
    <ul uib-dropdown-menu role="menu">
      <li>
        <a ng-click="$close(true)">Confirm and skip command view</a>
      </li>
    </ul>
  </div>
  <button class="btn btn-danger" ng-click="$dismiss('cancel')">Cancel <i class="fa fa-times-circle-o"></i></button>
</div>
