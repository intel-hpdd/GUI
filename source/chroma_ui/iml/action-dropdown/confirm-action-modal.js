//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2014 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.


angular.module('action-dropdown-module')
  .controller('ConfirmActionModalCtrl', ['$scope', 'title', 'confirmPrompts',
    function ConfirmActionModalCtrl  ($scope, title, confirmPrompts) {
      'use strict';

      $scope.confirmAction = {
        title: title,
        confirmPrompts: confirmPrompts
      };
    }])
  .factory('openConfirmActionModal', ['$modal', function openConfirmActionModalFactory ($modal) {
    'use strict';

    return function openConfirmActionModal (title, confirmPrompts) {
      return $modal.open({
        templateUrl: 'iml/action-dropdown/assets/html/confirm-action-modal.html',
        controller: 'ConfirmActionModalCtrl',
        windowClass: 'confirm-action-modal',
        backdropClass: 'confirm-action-modal-backdrop',
        backdrop: 'static',
        resolve: {
          title: _.fidentity(title),
          confirmPrompts: _.fidentity(confirmPrompts)
        }
      });
    };
  }]);
