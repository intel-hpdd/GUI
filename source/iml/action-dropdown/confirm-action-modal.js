//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import { Modal, Header, Body, Footer, Backdrop } from "../modal.js";
import WindowClickListener from "../window-click-listener.js";
import { linkEvent } from "inferno";
import DropdownComponent from "../dropdown-component.js";

import type { HighlandStreamT } from "highland";

const Confirmation = ({ confirmPrompts }) => {
  if (confirmPrompts.length === 1)
    return (
      <div class="single-confirmation">
        <p>{confirmPrompts[0]}</p>
      </div>
    );
  else if (confirmPrompts.length > 1)
    return (
      <div class="multi-confirmation">
        <h4>This action will:</h4>
        <ul class="well">
          {confirmPrompts.map(message => (
            <li>{message}</li>
          ))}
        </ul>
      </div>
    );
  else if (confirmPrompts.length === 0)
    return (
      <div>
        <p>Are you sure you want to perform this action?</p>
      </div>
    );
};

const onConfirm = ({ cb }) => {
  cb(ConfirmActionStates.CONFIRM);
};

const onConfirmAndSkip = ({ cb }) => {
  cb(ConfirmActionStates.CONFIRM_AND_SKIP);
};

const onCancel = ({ cb }) => {
  cb(ConfirmActionStates.CANCEL);
};

export const ConfirmActionStates = {
  CANCEL: "cancel",
  CONFIRM: "confirm",
  CONFIRM_AND_SKIP: "confirm_and_skip"
};

type ConfirmActionModalProps = {
  message: string,
  prompts: string[],
  cb: (string, () => HighlandStreamT<*>) => void
};

export const ConfirmActionModal = ({ message, prompts, cb }: ConfirmActionModalProps) => {
  return (
    <div id="confirm-modal" class="modal-open">
      <Modal visible={true} moreClasses={["confirm-action-modal"]}>
        <Header class="modal-header">
          <h4 class="modal-title">Confirm {message}</h4>
        </Header>
        <Body>
          <Confirmation confirmPrompts={prompts} />
        </Body>
        <Footer class="modal-footer">
          <div class="btn-group">
            <button type="button" class="btn btn-success" onClick={linkEvent({ cb }, onConfirm)}>
              Confirm <i class="fa fa-check-circle-o" />
            </button>
            <WindowClickListener>
              <DropdownComponent>
                <button
                  type="button"
                  class="btn btn-success dropdown-toggle"
                  aria-haspopup="true"
                  aria-expanded="false"
                >
                  <span class="caret" />
                  <span class="sr-only">Split button</span>
                </button>
                <ul role="menu" class="dropdown-menu">
                  <li>
                    <a onClick={linkEvent({ cb }, onConfirmAndSkip)}>Confirm and skip command view</a>
                  </li>
                </ul>
              </DropdownComponent>
            </WindowClickListener>
          </div>

          <button class="btn btn-danger" onClick={linkEvent({ cb }, onCancel)}>
            Cancel <i class="fa fa-times-circle-o" />
          </button>
        </Footer>
      </Modal>
      <Backdrop visible={true} moreClasses={["confirm-action-modal-backdrop"]} />
    </div>
  );
};
