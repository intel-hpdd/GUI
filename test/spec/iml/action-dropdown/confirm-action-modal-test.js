// @flow

import { ConfirmActionModal } from "../../../../source/iml/action-dropdown/confirm-action-modal.js";
import { render } from "inferno";

describe("confirm action modal", () => {
  let body, div, cb, button;
  beforeEach(() => {
    cb = jest.fn();

    div = document.createElement("div");
    body = document.querySelector("body");
    if (body) body.appendChild(div);
  });

  describe("with no prompts", () => {
    beforeEach(() => {
      render(<ConfirmActionModal message={"message"} prompts={[]} cb={cb} />, div);
    });

    it("should render the modal", () => {
      expect(div).toMatchSnapshot();
    });
  });

  describe("with single prompt", () => {
    beforeEach(() => {
      render(<ConfirmActionModal message={"message"} prompts={["prompt1"]} cb={cb} />, div);
    });

    it("should render the modal", () => {
      expect(div).toMatchSnapshot();
    });
  });

  describe("with multiple prompts", () => {
    beforeEach(() => {
      render(<ConfirmActionModal message={"message"} prompts={["prompt1", "prompt2"]} cb={cb} />, div);
    });

    it("should render the modal", () => {
      expect(div).toMatchSnapshot();
    });
  });

  describe("interaction", () => {
    beforeEach(() => {
      render(<ConfirmActionModal message={"message"} prompts={["prompt1"]} cb={cb} />, div);
    });

    describe("Confirming", () => {
      beforeEach(() => {
        button = div.querySelector(".modal-footer button:first-of-type");
        if (button) button.click();
      });

      it("should invoke the callback with confirm type", () => {
        expect(cb).toHaveBeenCalledTimes(1);
        expect(cb).toHaveBeenCalledWith("confirm");
      });
    });

    describe("Confirming and skipping", () => {
      let a;
      beforeEach(() => {
        button = div.querySelector(".modal-footer button.dropdown-toggle");
        if (button) button.click();
        a = div.querySelector(".modal-footer a");
        if (a) a.click();
      });

      it("should invoke the callback with confirm and skip type", () => {
        expect(cb).toHaveBeenCalledTimes(1);
        expect(cb).toHaveBeenCalledWith("confirm_and_skip");
      });
    });

    describe("Canceling", () => {
      beforeEach(() => {
        button = div.querySelector(".modal-footer button.btn-danger");
        if (button) button.click();
      });

      it("should invoke the callback with cancel type", () => {
        expect(cb).toHaveBeenCalledTimes(1);
        expect(cb).toHaveBeenCalledWith("cancel");
      });
    });
  });
});
