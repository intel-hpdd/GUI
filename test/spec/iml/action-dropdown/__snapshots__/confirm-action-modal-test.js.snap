// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirm action modal with multiple prompts should render the modal 1`] = `
<div>
  <div
    class="modal-open"
    id="confirm-modal"
  >
    <div
      class="modal fade in confirm-action-modal"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h4
              class="modal-title"
            >
              Confirm 
              message
            </h4>
          </div>
          <div
            class="modal-body "
          >
            <div
              class="multi-confirmation"
            >
              <h4>
                This action will:
              </h4>
              <ul
                class="well"
              >
                <li>
                  prompt1
                </li>
                <li>
                  prompt2
                </li>
              </ul>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-success"
                type="button"
              >
                Confirm 
                <i
                  class="fa fa-check-circle-o"
                />
              </button>
              <div
                class="btn-group dropdown "
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="btn btn-success dropdown-toggle"
                  type="button"
                >
                  <span
                    class="caret"
                  />
                  <span
                    class="sr-only"
                  >
                    Split button
                  </span>
                </button>
                <ul
                  class="dropdown-menu"
                  role="menu"
                >
                  <li>
                    <a>
                      Confirm and skip command view
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <button
              class="btn btn-danger"
            >
              Cancel 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in confirm-action-modal-backdrop"
    />
  </div>
</div>
`;

exports[`confirm action modal with no prompts should render the modal 1`] = `
<div>
  <div
    class="modal-open"
    id="confirm-modal"
  >
    <div
      class="modal fade in confirm-action-modal"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h4
              class="modal-title"
            >
              Confirm 
              message
            </h4>
          </div>
          <div
            class="modal-body "
          >
            <div>
              <p>
                Are you sure you want to perform this action?
              </p>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-success"
                type="button"
              >
                Confirm 
                <i
                  class="fa fa-check-circle-o"
                />
              </button>
              <div
                class="btn-group dropdown "
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="btn btn-success dropdown-toggle"
                  type="button"
                >
                  <span
                    class="caret"
                  />
                  <span
                    class="sr-only"
                  >
                    Split button
                  </span>
                </button>
                <ul
                  class="dropdown-menu"
                  role="menu"
                >
                  <li>
                    <a>
                      Confirm and skip command view
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <button
              class="btn btn-danger"
            >
              Cancel 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in confirm-action-modal-backdrop"
    />
  </div>
</div>
`;

exports[`confirm action modal with single prompt should render the modal 1`] = `
<div>
  <div
    class="modal-open"
    id="confirm-modal"
  >
    <div
      class="modal fade in confirm-action-modal"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h4
              class="modal-title"
            >
              Confirm 
              message
            </h4>
          </div>
          <div
            class="modal-body "
          >
            <div
              class="single-confirmation"
            >
              <p>
                prompt1
              </p>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-success"
                type="button"
              >
                Confirm 
                <i
                  class="fa fa-check-circle-o"
                />
              </button>
              <div
                class="btn-group dropdown "
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="btn btn-success dropdown-toggle"
                  type="button"
                >
                  <span
                    class="caret"
                  />
                  <span
                    class="sr-only"
                  >
                    Split button
                  </span>
                </button>
                <ul
                  class="dropdown-menu"
                  role="menu"
                >
                  <li>
                    <a>
                      Confirm and skip command view
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <button
              class="btn btn-danger"
            >
              Cancel 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in confirm-action-modal-backdrop"
    />
  </div>
</div>
`;
